tools/mason/MasonNew.chpl

```diff
@@ -67,64 +92,66 @@ proc masonNew(args) throws {
   }
 }
 
-proc validatePackageName(name) throws {
-  if name == '' {
+proc validatePackageName(dirName) throws {
+  if dirName == '' {
     throw new owned MasonError("No package name specified");
   }
-  else if !isIdentifier(name) {
-    throw new owned MasonError("Bad package name '" + name +
-                        "' - only Chapel identifiers are legal package names");
+  else if !isIdentifier(dirName) {
+    throw new owned MasonError("Bad package name '" + dirName +
+                        "' - only Chapel identifiers are legal package names.\n" +  
+                        "Please use mason new <illegal-name> --name <legal-name>");
   }
-  else if name.count("$") > 0 {
-    throw new owned MasonError("Bad package name '" + name +
+  else if dirName.count("$") > 0 {
+    throw new owned MasonError("Bad package name '" + dirName +
                         "' - $ is not allowed in package names");
   }
   else {
     return true;
   }
 }
 
-proc InitProject(name, vcs, show) throws {
+proc InitProject(dirName, packageName, vcs, show) throws {
```

Comment:
I know you didn't introduce these functions, but bonus points if you add some short doc-strings to them (following the mantra of "cleaning up code you modify").